src/main/java/games/wonders7/actions/BuyResourceR.java

```diff
@@ -0,0 +1,100 @@
+package games.wonders7.actions;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.actions.DrawCard;
+import games.wonders7.Wonders7Constants;
+import games.wonders7.Wonders7GameState;
+import games.wonders7.cards.Wonder7Card;
+
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+
+public class BuyResourceR extends DrawCard {
+
+    public String cardName;
+
+    // Player chooses card to play
+    public BuyResourceR(String cardName){
+        super();
+        this.cardName = cardName;
+
+    }
+
+
+    @Override
+    public boolean execute(AbstractGameState gameState) {
```

Comment:
Ouch - a lot of code duplicated between BuyResourceR and BuyResourceL. Would it be straightforward to make L/R a parameter?